app/models/audit_results.rb

```diff
@@ -113,66 +105,36 @@ def remove_db_updated_results
     result_array.delete_if { |res_hash| DB_UPDATED_CODES.include?(res_hash.keys.first) }
   end
 
-  def report_results
-    # Report completed
-    report_completed(completed_results)
-
-    # Report errors
-    report_errors(error_results)
-    result_array
+  def results
+    result_array.dup.freeze
   end
 
   def completed_results
-    result_array.select(&result_ok?)
+    result_array.select { |result| status_changed_to_ok?(result) }.freeze
   end
 
   def error_results
-    result_array.reject(&result_ok?)
+    result_array.reject { |result| status_changed_to_ok?(result) }.freeze
   end
 
   def contains_result_code?(code)
     result_array.detect { |result_hash| result_hash.key?(code) } != nil
   end
 
-  def status_changed_to_ok?(result)
-    /to ok$/.match(result[AuditResults::CM_STATUS_CHANGED]) != nil
-  end
-
   def results_as_string
     "#{string_prefix} #{result_array.map(&:values).flatten.join(' && ')}"
   end
 
   def to_json(*_args)
-    { druid: druid, result_array: result_array }.to_json
+    { druid: druid, results: result_array }.to_json
   end
 
   private
 
-  def result_ok?
-    proc { |result| status_changed_to_ok?(result) }
-  end
-
-  def reporters
-    @reporters ||= [
-      Reporters::LoggerReporter.new(logger),
-      Reporters::HoneybadgerReporter.new,
-      Reporters::AuditWorkflowReporter.new,
-      Reporters::EventServiceReporter.new
-    ].freeze
-  end
-
-  def report_errors(results)
-    reporters.each do |reporter|
-      reporter.report_errors(druid: druid, version: actual_version, storage_area: storage_area, check_name: check_name, results: results)
-    end
-  end
+  attr_reader :result_array
 
-  def report_completed(results)
-    reporters.each do |reporter|
-      results.each do |result|
-        reporter.report_completed(druid: druid, version: actual_version, storage_area: storage_area, check_name: check_name, result: result)
-      end
-    end
+  def status_changed_to_ok?(result)
+    /to ok$/.match(result[AuditResults::CM_STATUS_CHANGED]).present?
```

Comment:
If we're merely checking for the presence of a match, without caring about the value of the match, I think this method could be:

```suggestion
  def status_changed_to_ok?(result)
    /to ok$/.match?(result[AuditResults::CM_STATUS_CHANGED])
```

Reply:
True, but I would rather the method return a boolean instead of a Matcher.

Reply:
@justinlittman `#match?` returns a boolean. (`#match` doesn't.)

Reply:
Super. I'll make that change.